src/main/scala/org/deepdive/ddlog/DeepDiveLogCompiler.scala

```diff
@@ -130,18 +129,19 @@ class CompilationState( statements : DeepDiveLog.Program, config : DeepDiveLog.C
   }
 
   // Given a statement, resolve its name for the compiled extractor block.
-  def resolveExtractorBlockName(s: Statement): String = {
-    s match {
-      case s: SchemaDeclaration => s"extraction_rule_${statements indexOf s}"
-      case s: FunctionCallRule  => s"extraction_rule_${statements indexOf s}"
-      case s: ExtractionRule    => s"extraction_rule_${statements indexOf s}"
-      case s: InferenceRule     => s"extraction_rule_${s.q.head.name}"
-    }
-  }
+  def resolveExtractorBlockName(s: Statement): String = s"extraction_rule_${statements indexOf s}"
+  // {
+  //   s match {
+  //     case s: SchemaDeclaration => s"extraction_rule_${statements indexOf s}"
+  //     case s: FunctionCallRule  => s"extraction_rule_${statements indexOf s}"
+  //     case s: ExtractionRule    => s"extraction_rule_${statements indexOf s}"
+  //     case s: InferenceRule     => s"extraction_rule_${statements indexOf s}"
+  //   }
+  // }
```

Comment:
Could you remove these commented lines?

Ideally, we should use mnemonic names for each compiled extractor, so user's handwritten extractors can depend on them using it.  Is it easy to generate a unique but stable identifier for each?  This might be easy if we're already grouping by head and producing a union anyway.